### PR TITLE
EZP-32164: Implemented IconPathResolver service

### PR DESCRIPTION
**Pull request by webhdx**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-32164](https://jira.ez.no/browse/EZP-32164)
| **Improvement**| no
| **New feature**    | yes
| **Target version** | `2.2`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverInterface.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Platform\Assets\Resolver;
+
+interface IconPathResolverInterface
+{
+    public function resolve(string $icon, string $set = null): string;
```
Comment (alongosz) [inline src/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverInterface.php]:
```suggestion
    public function resolve(string $icon, ?string $set = null): string;
```

**Review by alongosz — CHANGES_REQUESTED** [review]
It should be quite quick to add simple unit test coverage for the implementation, just one method (with and without 2nd argument). Otherwise looks good.

CS remark:

**Comment by webhdx** [comment]
@alongosz Always feels great to remind yourself about your good ol' friend **unit tests**: d2efa7b

image

**Comment by webhdx** [comment]
Travis should be green in a minute

File: tests/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverTest.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+     * @dataProvider resolveDataProvider
+     */
+    public function testResolve(string $icon, ?string $set, string $expectedPath): void
+    {
+        $iconPathResolver = new IconPathResolver($this->configResolver, $this->packages);
+
+        self::assertEquals($expectedPath, $iconPathResolver->resolve($icon, $set));
+    }
+
+    public function resolveDataProvider()
```
Comment (alongosz) [inline tests/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverTest.php]:
```suggestion
    public function resolveDataProvider(): array
```

**Review by alongosz — APPROVED** [review]
Thx, I know it was a painful experience! ;) Looks good.
Minor CS:

File: tests/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverTest.php (hunk 2 above)
Comment (webhdx) [reply tests/IbexaPlatformAssetsBundle/lib/Resolver/IconPathResolverTest.php]: Fixed: 557d922

**Review by bogusez — APPROVED** [review]

**Pull request merged**